pkg/controller/hyperconverged/hyperconverged_controller.go

```diff
@@ -294,6 +313,11 @@ func (r *ReconcileHyperConverged) Reconcile(request reconcile.Request) (reconcil
 		}
 	}
 
+	// If the current image is not updated in CR ,then we're updating. This is also works when updating from
+	// an old version, since instance.ObjectMeta.Annotations[operatorImageCr] will be empty.
+	knownHcoVersion, _ := instance.Status.GetVersion(hcoVersionName)
+	r.upgradeMode = !init && knownHcoVersion != r.ownVersion
```

Comment:
please do this with an if so that we can eventually also log that we detected that we are in upgrade mode